### PR TITLE
Fixes #19700 - update rubocop rules

### PR DESCRIPTION
**Pull request by ohadlevy**
The following changes have been made:

- Performance/RedundantMerge:
changes lines such as:
```ruby
not_found_message.merge! :message => options
```
to:
```ruby
not_found_message[:message] = options
```

- converts str.match() to str =~ ()
- remove extra whitespaces etc.

### CODE REVIEW

**Review by dLobatog — CHANGES_REQUESTED** [review]
Thanks @ohadlevy ! Take a look at the seeds changes, it looks like tests are failing because of some problem there (including some Katello tests, only 11 of those failures are "supposed to be"  there)

File: db/seeds.d/08-data.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-SEEDED_PARTITION_TABLES = [
+SEEDED_PARTITION_TABLES ||= [
```
Comment (lzap) [inline db/seeds.d/08-data.rb]: Reason for this?

File: lib/proxy_api/dhcp.rb [hunk 2]
```diff
@@ -22,9 +22,12 @@ def subnet(subnet)
...
     def unused_ip(subnet, mac = nil)
       params = {}
-      params.merge!({:mac => mac}) if mac.present?
+      params[:mac] = mac if mac.present?
 
-      params.merge!({:from => subnet.from, :to => subnet.to}) if subnet.from.present? && subnet.to.present?
+      if subnet.from.present? && subnet.to.present?
+        params[:from] = subnet.from
+        params[:to] = subnet.to
+      end
```
Comment (lzap) [inline lib/proxy_api/dhcp.rb:30]: This might be reasonable case when to use `merge` tho, but I can live with that enforcing.

**Review by lzap — DISMISSED** [review]
Test failures, also I'd suggest setting `MaxKeyValuePairs` to just `1` - I think two lines of code is worth merging into one `merge` call. Defaults to 2.

File: lib/proxy_api/dhcp.rb (hunk 2 above)
Comment (lzap) [reply lib/proxy_api/dhcp.rb:30]: Also see my review comment, I'd change MaxKeyValuePairs.

File: db/seeds.d/08-data.rb (hunk 1 above)
Comment (ohadlevy) [reply db/seeds.d/08-data.rb]: I was trying to address ```foreman/db/seeds.d/08-data.rb:1: warning: already initialized constant SEEDED_PARTITION_TABLES``` I'll revert as its not really related to this PR.

**Comment by ohadlevy** [comment]
Tests are passing.

File: db/seeds.d/08-data.rb (hunk 1 above)
Comment (lzap) [reply db/seeds.d/08-data.rb]: Hmm looks we load this twice.

**Comment by ohadlevy** [comment]
@dLobatog can you re-review please? thanks.

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ohadlevy & @lzap !

**Pull request merged**